[JENKINS-29501] added support for crittercism plugin

marc-guenther opened the pull request:
added support for the Crittercism dSYM Plugin

daspilker commented:
I have been rejecting pull requests which add DSL methods for setting plain text passwords. In this case the API key is plain text. I started a discussion on the mailing list about this, see https://groups.google.com/d/msg/job-dsl-plugin/59xhodIgrsY/L2_GrZNEBQAJ.

daspilker commented on job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy:
```diff
@@ -1403,6 +1403,21 @@ class PublisherContext extends AbstractExtensibleContext {
     /**
+     * @since 1.36
+     */
+    @RequiresPlugin(id = 'crittercism-dsym')
```
Add a minimum version.

daspilker commented on docs/Job-reference.md:
````diff
@@ -5544,6 +5545,38 @@ job('example') {
+```
+
+Adds a Crittercism dSYM Upload publisher. Requires the [Crittercism dSYM Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Crittercism+dSYM+Plugin).
+
````
Add a link to the new "Handling Credentials" page to give the user a hint about handling plain text credentials. See https://github.com/jenkinsci/job-dsl-plugin/blob/e917f45419ca2cbcf4cb95e673acdda6667eeeef/docs/Job-reference.md#hipchat-publisher for an example.

daspilker commented:
Please rebase you changes to the current master branch.

marc-guenther commented:
Just rebased, added the minimal plugin version (there has been only one release of the plugin so far, though), and added the link to the "Handling Credentials" page.

daspilker commented:
Thanks! I had to rebased you changes again and merged them manually.
